### PR TITLE
[minigraph] add support for changing T1 ports speed from 400G to 100G and vice-versa

### PR DESCRIPTION
**Pull request by vdahiya12**
On SONiC T1 cisco 8101 HwSku, the speed changes are done from 400G to 100G needs to be supported on 400G ports.
To enable this, along with speed change the port lanes need to be changed. This PR has the changes to update the port lanes when such speed change happens.
Basically if Banwidth in minigraph.xml intends to enable a 100G speed on a 400G port, then the appropriate lane change and speed change needs to be invoked in mingraph parser
Example if port_config.ini dicatates the speed to be 400G and minigraph has 100G speed, then this changeneeds to be accommodated
```
# name         lanes                                      alias   index  speed    channel
Ethernet96     1536,1537,1538,1539,1540,1541,1542,1543    etp12    12       400000     0
```

```
 <DeviceLinkBase>
        <ElementType>DeviceInterfaceLink</ElementType>
        <EndDevice>ARISTA01T2</EndDevice>
        <EndPort>Ethernet1</EndPort>
        <StartDevice>Device-8101-01</StartDevice>
        <StartPort>etp12</StartPort>
        <Bandwidth>100000</Bandwidth>
      </DeviceLinkBase>
```

These platforms today have 400g port with 8 serdes lines, and 100g will operate with 4 serdes lane. When the port speed changes from 400G to 100G the first 4 lanes will be used for 100G port.

.

#### How I did it
Updated the port lanes when the port speed is changed from 400g to 100g
#### How to verify it
UT and tested on a 8101 - DUT 
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

### CODE REVIEW

**Comment by prgeor** [comment]
@vdahiya12 can you add unit test

File: src/sonic-config-engine/minigraph.py [hunk 1]
```diff
@@ -1679,7 +1679,8 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
         port_default_speed =  port_speeds_default.get(port_name, None)
         port_png_speed = port_speed_png[port_name]
 
-        if switch_type == 'voq':
+        # Add a check for for voq, T1
+        if switch_type in ['voq', 'LeafRouter']:
```
Comment (arlakshm) [inline src/sonic-config-engine/minigraph.py:1683]: switch_type will not be 'leafRouter'  you can use this check `results['DEVICE_METADATA']['localhost']['type'].lower() == 'leafrouter'` to check of the device is T1

**Comment by prgeor** [comment]
@vdahiya12 do you still want PR to be in DRAFT?

File: src/sonic-config-engine/minigraph.py (hunk 1 above)
Comment (vdahiya12) [reply src/sonic-config-engine/minigraph.py]: fixed

**Comment by vdahiya12** [comment]
> @vdahiya12 can you add unit test

added

**Review by prgeor — DISMISSED** [review]

**Review by lolyu — COMMENTED** [review]
The new intro of `channel` key in port table  will break the port table yang: https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-port.yang, which leads to your UT failure

**Review by arlakshm — APPROVED** [review]
lgtm

**Review by prgeor — APPROVED** [review]
@lguohan please help merge

**Comment by prgeor** [comment]
@StormLiangMS @yxieca please help cherry pick

**Comment by yxieca** [comment]
@vdahiya12 can you create separate PR for 202205 branch?

**Pull request merged**